File: eCommerce/eCommerce.Domain/Models/Product.cs
﻿using eCommerce.Domain.Models.Base;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace eCommerce.Domain.Models
{
    public class Product : BaseModel
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public int Department { get; set; }

        public virtual User User { get; set; }

        public int UserId { get; set; }

        public int Amount { get; set; }

        public double Price { get; set; }

        public virtual List<OrderProduct> OrderProducts { get; set; }

        public virtual List<ProductImage> ProductImages { get; set; }
    }
}

File: eCommerceFront/src/app/product/product.ts
import { FileHandle } from "./file-handler";

export interface Product{
    
    id?: number;
    name?: string;
    description?: string;
    department?: number;
    userId?: number;
    amount?: number;
    price?: number;
    productImages?: FileHandle[];
    [key: string]: any;
}

export interface ProductState{
    products: Product[]
}